internal/conversion.go

```diff
@@ -0,0 +1,78 @@
+package internal
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/hazelcast/hazelcast-go-client/serialization"
+)
+
+// supported types
+const (
+	String  = "string"
+	Boolean = "boolean"
+	JSON    = "json"
+	Int8    = "int8"
+	Int16   = "int16"
+	Int32   = "int32"
+	Int64   = "int64"
+	Float32 = "float32"
+	Float64 = "float64"
+)
+
+var SupportedTypes = []string{
+	String,
+	Boolean,
+	JSON,
+	Int8,
+	Int16,
+	Int32,
+	Int64,
+	Float32,
+	Float64,
+}
+
+func ConvertString(value, valueType string) (interface{}, error) {
+	var (
+		convertedValue interface{}
+		err            error
+	)
+	var (
+		intermediateInt   int64
+		intermediateFloat float64
+	)
+	switch valueType {
+	case String:
+		convertedValue = value
+	case Boolean:
+		convertedValue, err = strconv.ParseBool(value)
+	case JSON:
+		convertedValue = serialization.JSON(value)
```

Comment:
I think it would be nice to check whether the given JSON is valid by trying to unmarshal it.

Reply:
JSON validation on https://github.com/hazelcast/hazelcast-commandline-client/pull/71/commits/e1b36476184397d20088747e12f583a767f096b1